### PR TITLE
NEW: table order logic and endpoint

### PR DESCRIPTION
**Pull request by SecondeJK**

### CODE REVIEW

File: tests/Feature/Api/UpdateTableNumberTest.php [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+     */
+    public function cannotUpdateOrderWithoutTableService(): void
+    {
+        $merchant = $this->createAndReturnMerchant();
+        $originalTableNumber = $this->faker->numberBetween(0, 35);
+        $order = $this->createAndReturnOrderForStatus(
+            'Purchased Order',
+            [
+                'merchant_id' => $merchant->id,
+                'table_number' => $originalTableNumber,
```
Comment (dannewnsjump24) [inline tests/Feature/Api/UpdateTableNumberTest.php:154]: We should probably use null here as its an order without table service so it really shouldnt have a table number in the database.

File: app/Order.php [hunk 2]
```diff
@@ -513,6 +515,14 @@ public function hasPaymentDetails(): bool
         return $this->payment_id !== null;
     }
 
+    /**
+     * @return bool
+     */
+    public function isTableService(): bool
```
Comment (dannewnsjump24) [inline app/Order.php:521]: Can we just get some tests on this to make sure we handle `null` `true` and `false` in the DB

Comment (AlexJump24) [reply app/Order.php:521]: Agreed as simple as the test will be. Although I think its great you have moved this onto a function on the model. Which will make things a lot more readable long term and potentially more flexible if the condition ever evolved.

File: tests/Unit/Service/OrderServiceTest.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            'orders',
+            [
+                'id' => $order->id,
+                'table_number' => null
+            ]
+        );
+
+        $tableNumber = $this->faker->numberBetween(0, 20);
+        $return = $this->orderService->setTableNumberOnOrder($order, $tableNumber);
+        $this->assertEquals(false, $return);
```
Comment (AlexJump24) [inline tests/Unit/Service/OrderServiceTest.php:104]: Update to assertFalse. 

- [ ] Change to assertFalse.

File: tests/Unit/Service/OrderServiceTest.php [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            'orders',
+            [
+                'id' => $order->id,
+                'table_number' => null
+            ]
+        );
+
+        $this->expectException(\TypeError::class);
+        $return = $this->orderService->setTableNumberOnOrder($order, 'boo!');
+        $this->assertEquals(false, $return);
```
Comment (AlexJump24) [inline tests/Unit/Service/OrderServiceTest.php:76]: Update to assertFalse like below.

- [ ] Change to assertFalse()

File: tests/Feature/Api/UpdateTableNumberTest.php [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        ]);
+    }
+
+    /**
+     * @test
+     */
+    public function cannotUpdateTableNumberWithString(): void
+    {
+        $merchant = $this->createAndReturnMerchant();
+        $originalTableNumber = $this->faker->numberBetween(0, 35);
```
Comment (AlexJump24) [inline tests/Feature/Api/UpdateTableNumberTest.php:109]: Just a suggestion and you don't have to agree, but you could add this to the Orders factory as a single state.

E.g. 'Table Number'. Then in your tests you can do factory(Order::class)->state('Table Number')->raw() (raw never persists the data to the database) which should return the table number and therefore centralise the code. 

Or you may be able to do an only on the factory however I am not 100% sure if it needlessly processes other fake data generation then.

**Pull request merged**